gazebo/common/SphericalCoordinates_TEST.cc

```diff
@@ -261,9 +343,26 @@ TEST_F(SphericalCoordinatesTest, Distance)
   longA.Degree(-122.249972);
   latB.Degree(46.124953);
   longB.Degree(-122.251683);
-  double d = common::SphericalCoordinates::Distance(latA, longA, latB, longB);
 
-  EXPECT_NEAR(14002, d, 20);
+  // Calculating distance using the static method.
+  double d1 = common::SphericalCoordinates::Distance(latA, longA, latB, longB);
+  EXPECT_NEAR(14002, d1, 20);
+
+  // Using the non static method. The default surface type is EARTH_WGS84.
+  common::SphericalCoordinates earthSC = common::SphericalCoordinates();
```

Comment:
I'm seeing a test failure in CI and I think it's related to this line

https://build.osrfoundation.org/job/gazebo-ci-pr_any-ubuntu_auto-amd64-gpu-nvidia/1002/testReport/junit/(root)/UNIT_SphericalCoordinates_TEST/test_ran/

~~~
53: [----------] 5 tests from SphericalCoordinatesTest
53: [ RUN      ] SphericalCoordinatesTest.Constructor
53: [       OK ] SphericalCoordinatesTest.Constructor (0 ms)
53: [ RUN      ] SphericalCoordinatesTest.Convert
53: [       OK ] SphericalCoordinatesTest.Convert (3 ms)
53: [ RUN      ] SphericalCoordinatesTest.SetFunctions
53: [       OK ] SphericalCoordinatesTest.SetFunctions (0 ms)
53: [ RUN      ] SphericalCoordinatesTest.CoordinateTransforms
53: [       OK ] SphericalCoordinatesTest.CoordinateTransforms (0 ms)
53: [ RUN      ] SphericalCoordinatesTest.Distance
53: free(): double free detected in tcache 2
 53/336 Test  #53: UNIT_SphericalCoordinates_TEST ...............***Exception: Child aborted  0.32 sec
~~~

I found a backtrace with `gdb` and it seemed to emanate from this line

Reply:
Tested locally, fixed https://github.com/osrf/gazebo/pull/3250/commits/e1109c186ef8dd2cd77bbd9d9b20adf2b00afad4